File: backend_java_spring_boot/myPizza/src/main/java/backendapp/myPizza/controllers/OrderController.java
package backendapp.myPizza.controllers;

import backendapp.myPizza.Models.entities.Order;
import backendapp.myPizza.Models.entities.OrderSet;
import backendapp.myPizza.Models.entities.TimeInterval;
import backendapp.myPizza.Models.reqDTO.ConfirmOrderDTO;
import backendapp.myPizza.Models.resDTO.ConfirmRes;
import backendapp.myPizza.exceptions.BadRequestException;
import backendapp.myPizza.exceptions.NotFoundException;
import backendapp.myPizza.exceptions.Validation;
import backendapp.myPizza.services.OrderService;
import jakarta.persistence.Access;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.validation.BindingResult;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping("/api/order")
public class OrderController {

    @Autowired
    private OrderService orderSvc;

    @PreAuthorize("hasAuthority('ADMIN')")
    @GetMapping("/{orderId}")
    public Order getOrderById(@PathVariable UUID orderId) throws BadRequestException {
        return orderSvc.getOrderById(orderId);
    }

    @PreAuthorize("hasAuthority('ADMIN')")
    @PostMapping("/confirm")
    public ConfirmRes confirmOrder(@RequestBody @Validated ConfirmOrderDTO confirmOrderDTO, BindingResult validation) throws BadRequestException {
        Validation.validate(validation);
        return orderSvc.confirmOrder(confirmOrderDTO.orderId(), confirmOrderDTO.timeIntervalId());
    }

    @PreAuthorize("hasAuthority('ADMIN')")
    @GetMapping("/{orderId}/reject")
    public ConfirmRes confirmOrder(@PathVariable UUID orderId) throws BadRequestException {
        return orderSvc.rejectOrder(orderId);
    }

    @PreAuthorize("hasAuthority('ADMIN')")
    @GetMapping("/{orderId}/set-completed")
    public List<TimeInterval> completeOrder(@PathVariable UUID orderId) throws BadRequestException, NotFoundException {
        return orderSvc.completeOrder(orderId);
    }

}

File: backend_nodejs_nest_fastify/my-pizza/src/nest_modules/order/controllers/order.controller.ts
import { Body, Controller, Get, Param, Post, UseGuards } from '@nestjs/common';
import { OrderService } from '../services/order.service';
import { AdminGuard } from 'src/nest_modules/auth-user/guards/admin.guard';
import { UUID } from 'crypto';
import { OrderRes } from '../interfaces/order-res-interface';
import { ConfirmOrderDTO } from '../interfaces/confirm-order-dto.interface';
import { ConfirmRes } from 'src/nest_modules/auth-user/interfaces/confirm-res.interface';
import { TimeInterval } from '../entities/time-interval.entity';

@Controller('api/order')
@UseGuards(AdminGuard)
export class OrderController {

    constructor(private orderSvc: OrderService) { }

    
    @Get('/:orderId')
    public async getOrderById(@Param('orderId') orderId: UUID): Promise<OrderRes> {
        return await this.orderSvc.getOrderById(orderId)
    }

    
    @Post('/confirm')
    public async confirmOrder(@Body() confirmOrderDTO: ConfirmOrderDTO): Promise<ConfirmRes> {
        return await this.orderSvc.confirmOrder(confirmOrderDTO.orderId, confirmOrderDTO.timeIntervalId)
    }

    @Get('/:orderId/reject')
    public async rejectOrder(@Param('orderId') orderId: UUID): Promise<ConfirmRes> {
        return await this.rejectOrder(orderId)
    }
    
    @Get('/:orderId/set-completed')
    public async completeOrder(@Param('orderId') orderId: UUID): Promise<TimeInterval[]> {
        return await this.orderSvc.completeOrder(orderId)
    }

}
